src/cdk/testing/protractor/protractor-harness-environment.ts

```diff
@@ -7,18 +7,24 @@
  */
 
 import {HarnessEnvironment, HarnessLoader, TestElement} from '@angular/cdk/testing';
-import {by, element as protractorElement, ElementFinder} from 'protractor';
+import {by, element as protractorElement, ElementArrayFinder, ElementFinder} from 'protractor';
 import {ProtractorElement} from './protractor-element';
 
+/** The default query function that respects shadow boundaries. */
+const defaultQueryFn = (selector: string, root: ElementFinder) => root.all(by.css(selector));
+
 /** A `HarnessEnvironment` implementation for Protractor. */
 export class ProtractorHarnessEnvironment extends HarnessEnvironment<ElementFinder> {
-  protected constructor(rawRootElement: ElementFinder) {
+  protected constructor(rawRootElement: ElementFinder,
+      private _queryFn: (selector: string, root: ElementFinder) => ElementArrayFinder =
+          defaultQueryFn) {
     super(rawRootElement);
   }
 
   /** Creates a `HarnessLoader` rooted at the document root. */
-  static loader(): HarnessLoader {
-    return new ProtractorHarnessEnvironment(protractorElement(by.css('body')));
+  static loader(queryFn?: (selector: string, root: ElementFinder) => ElementArrayFinder):
```

Comment:
We probably want to add a config object rather than passing the query impl directly in case we want to add more options in the future. I think putting it in a config also helps de-emphasize the custom querySelector impl.